shipyard/src/test/java/org/jclouds/shipyard/features/ServiceKeysApiMockTest.java

```diff
@@ -0,0 +1,101 @@
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+
+import org.jclouds.http.HttpResponseException;
+import org.jclouds.shipyard.ShipyardApi;
+import org.jclouds.shipyard.domain.servicekeys.CreateServiceKey;
+import org.jclouds.shipyard.domain.servicekeys.DeleteServiceKey;
+import org.jclouds.shipyard.domain.servicekeys.ServiceKey;
+import org.jclouds.shipyard.internal.BaseShipyardMockTest;
+import org.testng.annotations.Test;
+
+import com.squareup.okhttp.mockwebserver.MockResponse;
+import com.squareup.okhttp.mockwebserver.MockWebServer;
+
+/**
+ * Mock tests for the {@link org.jclouds.shipyard.features.ServiceKeysApi} class.
+ */
+@Test(groups = "unit", testName = "ServiceKeysApiMockTest")
+public class ServiceKeysApiMockTest extends BaseShipyardMockTest {
+
+   public void testListServiceKeys() throws Exception {
+      MockWebServer server = mockShipyardWebServer();
+      server.enqueue(new MockResponse().setResponseCode(200).setBody(payloadFromResource("/servicekeys.json")));
+      ShipyardApi shipyardApi = api(server.getUrl("/"));
+      ServiceKeysApi api = shipyardApi.serviceKeysApi();
+      try {
+         assertEquals(api.listServiceKeys().size(), 2);
+         assertSent(server, "GET", "/api/servicekeys");
+      } finally {
+         shipyardApi.close();
+         server.shutdown();
+      }
+   }
+
+   public void testCreateServiceKey() throws Exception {
+      MockWebServer server = mockShipyardWebServer();
+      server.enqueue(new MockResponse().setResponseCode(200).setBody(payloadFromResource("/servicekey-response.json")));
+      ShipyardApi shipyardApi = api(server.getUrl("/"));
+      ServiceKeysApi api = shipyardApi.serviceKeysApi();
+
+      CreateServiceKey createServiceKey = CreateServiceKey.create("jclouds-shipyard-testing");
+      try {
+         ServiceKey serviceKey = api.createServiceKey(createServiceKey);
+         assertNotNull(serviceKey);
+         assertEquals(serviceKey.description(), "jclouds-shipyard-testing");
+         assertSent(server, "POST", "/api/servicekeys", new String(payloadFromResource("/servicekey-create.json")));
+      } finally {
+         shipyardApi.close();
+         server.shutdown();
+      }
+   }
+
+   public void testDeleteServiceKey() throws Exception {
+      MockWebServer server = mockShipyardWebServer();
+      server.enqueue(new MockResponse().setResponseCode(204));
+      ShipyardApi shipyardApi = api(server.getUrl("/"));
+      ServiceKeysApi api = shipyardApi.serviceKeysApi();
+      try {
+         DeleteServiceKey deleteServiceKey = DeleteServiceKey.create("1111222233334444");
+         api.deleteServiceKey(deleteServiceKey);
+         assertSent(server, "DELETE", "/api/servicekeys");
+      } finally {
+         shipyardApi.close();
+         server.shutdown();
+      }
+   }
+   
+   @Test (expectedExceptions = HttpResponseException.class)
+   public void testDeleteNonExistentServiceKey() throws Exception {
```

Comment:
Don't add tests for paths that are not explicitly present in the code. If you don't declare a fallback, then remove the test. There's no need to test the jclouds default behavior.